File: Leetcode/1302/1302.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxDepth;
    int sum;
    void dfs1(TreeNode* root, int level) {
        if (root->left && root->right) {
            dfs1(root->left, level + 1);
            dfs1(root->right, level + 1);
        }
        else if (!root->left && root->right) {
            dfs1(root->right, level + 1);
        }
        else if (root->left && !root->right) {
            dfs1(root->left, level + 1);
        }
        else {
            maxDepth = max(maxDepth, level);
        }
    }
    void dfs2(TreeNode* root, int level) {
        if (root->left && root->right) {
            dfs2(root->left, level + 1);
            dfs2(root->right, level + 1);
        }
        else if (!root->left && root->right) {
            dfs2(root->right, level + 1);
        }
        else if (root->left && !root->right) {
            dfs2(root->left, level + 1);
        }
        else {
            if (level == maxDepth) {
                sum += root->val;
            }
        }
    }
    int deepestLeavesSum(TreeNode* root) {
        if (!root) return 0;
        maxDepth = -1;
        sum = 0;
        dfs1(root, 0);
        dfs2(root, 0);
        return sum;
    }
};

File: Leetcode/1302/1302.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
var (
	height int
	sum int
)
func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}
func getHeight(root *TreeNode) int {
	if root != nil {
		return 1 + max(getHeight(root.Left), getHeight(root.Right))
	}
	return 0
}
func dfs(root *TreeNode, currHeight int) {
	if height == currHeight {
		sum += root.Val
	}
	if root.Left != nil {
		dfs(root.Left, currHeight+1)
	}
	if root.Right != nil {
		dfs(root.Right, currHeight+1)
	}
}
func deepestLeavesSum(root *TreeNode) int {
	height = getHeight(root)
	sum = 0
	dfs(root, 1)
	return sum
}